python/lsst/ip/isr/isrMock.py

```diff
@@ -0,0 +1,915 @@
+        pattern should be visually identifiable.  The (x, y)
+        coordinates are in the frame of the amplifier, and (u, v) in
+        the frame of the full trimmed image.
+        """
+        for x in range(0, ampData.getDimensions().getX()):
+            for y in range(0, ampData.getDimensions().getY()):
+                (u, v) = self.localCoordToExpCoord(amp, x, y)
+                ampData.getArray()[y][x] = (ampData.getArray()[y][x] +
+                                            scale * np.sinc(((u - 100)/50)**2 + (v / 50)**2))
+
+    def amplifierAddFlat(self, amp, ampData, size):
+        r"""Add a flat-like dome pattern to an amplifier's image data.
+
+        Parameters
+        ----------
+        amp : `lsst.afw.ampInfo.AmpInfoRecord`
+            Amplifier to operate on.  Needed for amp<->exp coordinate transforms.
+        ampData : `lsst.afw.image.ImageF`
+            Amplifier image to operate on.
+        size : float
+            Size of the dome to use.  A larger value results in a
+            smaller center-to-edge fall off.
+
+        Notes
+        -----
+        This uses a centered 2-d Gaussian to simulate an illumination
+        pattern that falls off towards the edge of the detector.  The
+        e(x, y) coordinates are in the frame of the amplifier, and (u, v)
+        in the frame of the full trimmed image.
+        """
+        for x in range(0, ampData.getDimensions().getX()):
+            for y in range(0, ampData.getDimensions().getY()):
+                (u, v) = self.localCoordToExpCoord(amp, x, y)
+                f = np.exp(-0.5 * ((u - 100)**2 + (v - 100)**2) / size**2)
+                ampData.getArray()[y][x] = (ampData.getArray()[y][x] * f)
+
+
+class RawMock(IsrMock):
+    r"""Generate a raw exposure suitable for ISR.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.isTrimmed = False
+        self.config.doGenerateImage = True
+        self.config.doGenerateDict = False
+        self.config.doOverscan = True
+        self.config.doSky = True
+        self.config.doSource = True
+        self.config.doCrosstalk = False
+        self.config.doBias = True
+        self.config.doDark = True
+
+
+class TrimmedRawMock(RawMock):
+    r"""Generate a trimmed raw exposure.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.isTrimmed = True
+        self.config.doOverscan = False
+
+
+class CalibratedRawMock(RawMock):
+    r"""Generate a trimmed raw exposure.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.isTrimmed = True
+        self.config.doOverscan = False
+        self.config.doBias = True
+        self.config.doDark = False
+        self.config.doFlat = False
+        self.config.doFringe = False
+        self.config.doCrosstalk = True
+        self.config.biasLevel = 0.0
+        self.config.biasSigma = 0.0
+        self.config.skySigma = 100.0
```

Comment:
Right, this is exactly what I feared - changing the `skySigma` independent of the `skyLevel`. Does this makes sense?